test/main.cpp

```diff
@@ -514,6 +507,8 @@ EXPORT int main(int argc, char const* argv[])
 	if (redirect_stdout) fflush(stdout);
 	if (redirect_stderr) fflush(stderr);
 
-	return print_failures() ? 333 : 0;
+	int total_num_failures = print_failures();
+
+	return total_num_failures ? 333 : 0;
```

Comment:
is there a good reason for this change? If so, ``total_num_failures`` should be const